File: backend/app/services/export_service.py
from app.services.note_store import get_note_store


async def export_to_markdown(pdf_identifier: str) -> str:
    store = get_note_store()
    notes = await store.list_notes(pdf_identifier)

    if not notes:
        return "# Annotations\n\nNo annotations found for this document.\n"

    pdf_title = notes[0].get("pdf_title") or "Untitled Document"
    lines = [
        f"# Annotations: {pdf_title}",
        "",
        f"*Source: {pdf_identifier}*",
        "",
        "*Exported from PDF Converser*",
        "",
        "---",
        "",
    ]

    current_page = None
    for note in notes:
        if note.get("page_number", 0) != current_page:
            current_page = note.get("page_number", 0)
            page_label = f"Page {current_page}" if current_page > 0 else "Page (unknown)"
            lines.append(f"## {page_label}")
            lines.append("")

        type_label = note.get("comment_type", "summary").replace("_", " ").title()
        lines.append(f"**[{type_label}]**")
        lines.append("")
        lines.append(f"> {note['selected_text']}")
        lines.append("")
        lines.append(note["cleaned_comment"])
        lines.append("")
        lines.append("<details><summary>Raw voice note</summary>")
        lines.append("")
        lines.append(note["raw_transcript"])
        lines.append("")
        lines.append("</details>")
        lines.append("")
        lines.append("---")
        lines.append("")

    return "\n".join(lines)

File: services/export-service.js
// Markdown export. Port of backend/app/services/export_service.py.
// Output formatting is intentionally byte-identical to the Python version.

const { getNoteStore } = require('./note-store');

function titleCase(s) {
  return s.replace(/_/g, ' ').replace(/\b\w/g, (c) => c.toUpperCase());
}

async function exportToMarkdown(pdfIdentifier) {
  const store = getNoteStore();
  const notes = await store.listNotes(pdfIdentifier);

  if (notes.length === 0) {
    return '# Annotations\n\nNo annotations found for this document.\n';
  }

  const pdfTitle = notes[0].pdf_title || 'Untitled Document';
  const lines = [
    `# Annotations: ${pdfTitle}`,
    '',
    `*Source: ${pdfIdentifier}*`,
    '',
    '*Exported from PDF Converser*',
    '',
    '---',
    '',
  ];

  let currentPage = null;
  for (const note of notes) {
    const page = note.page_number || 0;
    if (page !== currentPage) {
      currentPage = page;
      const label = currentPage > 0 ? `Page ${currentPage}` : 'Page (unknown)';
      lines.push(`## ${label}`);
      lines.push('');
    }

    const typeLabel = titleCase(note.comment_type || 'summary');
    lines.push(`**[${typeLabel}]**`);
    lines.push('');
    lines.push(`> ${note.selected_text}`);
    lines.push('');
    lines.push(note.cleaned_comment);
    lines.push('');
    lines.push('<details><summary>Raw voice note</summary>');
    lines.push('');
    lines.push(note.raw_transcript);
    lines.push('');
    lines.push('</details>');
    lines.push('');
    lines.push('---');
    lines.push('');
  }

  return lines.join('\n');
}

module.exports = { exportToMarkdown };
